lib/portlayer/task/common.go

```diff
@@ -31,30 +31,38 @@ func toggleActive(op *trace.Operation, h interface{}, id string, active bool) (i
 	if !ok {
 		return nil, fmt.Errorf("Type assertion failed for %#+v", handle)
 	}
-
-	stasks := handle.ExecConfig.Sessions
-	etasks := handle.ExecConfig.Execs
-
-	taskS, okS := stasks[id]
-	taskE, okE := etasks[id]
-
-	if !okS && !okE {
-		return nil, fmt.Errorf("unknown task ID: %s", id)
+	taskS, okS := handle.ExecConfig.Sessions[id]
+	taskE, okE := handle.ExecConfig.Execs[id]
+
+	op.Debugf("target task ID: %s", id)
+	op.Debugf("session tasks during inspect: %s", handle.ExecConfig.Sessions)
+	// print all of them, otherwise we will have to assemble the id list regardless of
+	// the log level at the moment. If there is a way to check the log level we should
+	// do that. since the other approach will slow down all calls to toggleActive.
+	op.Debugf("exec tasks during inspect: %s", handle.ExecConfig.Execs)
+
+	var task *executor.SessionConfig
+	if okS {
+		task = taskS
 	}
 
-	task := taskS
-	if handle.Runtime != nil && handle.Runtime.PowerState != types.VirtualMachinePowerStatePoweredOff {
-		op.Debugf("Task bind configuration applies to ephemeral set")
+	if okE {
 		task = taskE
+	}
 
+	// if no task has been joined that can be manipulated in the container's current state
+	if task == nil {
+		// FIXME return a compatibility style error here. Propagate it back to the user.
 		if err := compatible(handle); err != nil {
 			return nil, err
 		}
+
+		// NOTE: this was the previous error, before merging we need to decide which one to use.
+		// return nil, fmt.Errorf("Cannot modify task %s in current state", id)
+		return nil, TaskNotFoundError{msg: fmt.Sprintf("Cannot find task %s", id)}
 	}
 
-	// if no task has been joined that can be manipulated in the container's current state
 	if task == nil {
```

Comment:
Seems like this code makes no sense anymore.

Reply:
Your absolutely right. I was focused on the above case and forgot to remove this one. There still reamains a questing there for @hickeng as to whether or not we keep the old error message or we take the new one.

Reply:
good catche on the empty if statement, I am surprised that `govet` did not complain at all